cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -1752,4 +1786,111 @@ mod tests {
             panic!("expected LockForMigration error");
         }
     }
+
+    /// [CoreEnvironment::migrate_manifest_contents_to_v1] migrates a manifest
+    /// with `script` in a `[hook]` table correctly, maintaining comments and
+    /// formatting.
+    #[test]
+    fn migrate_script_hook_table() {
+        let contents = formatdoc! {r#"
+            [vars]
+            foo = "bar"
+
+            # comment 1
+            [hook] # comment 2
+            # comment 3
+             script = "echo hello" # comment 4
+            # comment 5
+
+            [options]
+            "#};
+        let mut raw_manifest = RawManifest::from_str(&contents).unwrap();
+        CoreEnvironment::migrate_manifest_contents_to_v1(&mut raw_manifest).unwrap();
+        assert_eq!(raw_manifest.to_string(), formatdoc! {r#"
+                version = 1
+                [vars]
+                foo = "bar"
+
+                # comment 1
+                [hook] # comment 2
+                # comment 3
+                 on-activate = "echo hello" # comment 4
+                # comment 5
+
+                [options]
+                "#
+        });
+    }
+
+    /// [CoreEnvironment::migrate_manifest_contents_to_v1] migrates a manifest
+    /// with hook.script as a dotted key correctly, maintaining comments and
+    /// formatting.
+    #[test]
+    fn migrate_script_hook_dotted_decor() {
+        let contents = formatdoc! {r#"
+            vars.foo = "bar"
+
+            # comment 1
+            hook . script = "echo hello" # comment 2
+            # comment 3
+
+            options.allow.unfree = false
+            "#};
+        let mut raw_manifest = RawManifest::from_str(&contents).unwrap();
+        CoreEnvironment::migrate_manifest_contents_to_v1(&mut raw_manifest).unwrap();
+        assert_eq!(raw_manifest.to_string(), formatdoc! {r#"
+                vars.foo = "bar"
+
+                # comment 1
+                hook . on-activate = "echo hello" # comment 2
+                # comment 3
+
+                options.allow.unfree = false
+                version = 1
+                "#
+        });
+    }
+
+    /// If a manifest contains both `hook.script` and `hook.on-activate`,
+    /// [CoreEnvironment::migrate_manifest_contents_to_v1] returns an error.
+    #[test]
+    fn migrate_script_skip_for_on_activate() {
+        let contents = formatdoc! {r#"
+            [hook]
+            script = "echo hello"
+            on-activate = "echo hello"
+            "#};
+        let mut raw_manifest = RawManifest::from_str(&contents).unwrap();
+        let err = CoreEnvironment::migrate_manifest_contents_to_v1(&mut raw_manifest).unwrap_err();
+        assert_eq!(raw_manifest.to_string(), formatdoc! {r#"
+                version = 1
+                [hook]
+                script = "echo hello"
+                on-activate = "echo hello"
```

Comment:
Nit: using different values here would make it clearer that we don't modify the original values in a partial migration.
```suggestion
            [hook]
            script = "echo foo"
            on-activate = "echo bar"
            "#};
        let mut raw_manifest = RawManifest::from_str(&contents).unwrap();
        let err = CoreEnvironment::migrate_manifest_contents_to_v1(&mut raw_manifest).unwrap_err();
        assert_eq!(raw_manifest.to_string(), formatdoc! {r#"
                version = 1
                [hook]
                script = "echo foo"
                on-activate = "echo bar"
```